pkg/core/mpt/billet.go

```diff
@@ -242,12 +242,32 @@ func (b *Billet) traverse(curr Node, path, from []byte, process func(pathToNode
 		var (
 			startIndex byte
 			endIndex   byte = childrenCount
+			inc             = func(i byte) byte {
+				i++
+				return i
+			}
+			cmp = func(i byte) bool {
+				return i < endIndex
+			}
 		)
+		if backwards {
+			startIndex, endIndex = endIndex, startIndex
+			inc = func(i byte) byte {
+				i--
+				return i
+			}
+			cmp = func(i byte) bool {
+				return i >= startIndex
+			}
+		}
 		if len(from) != 0 {
 			endIndex = lastChild
+			if backwards {
+				endIndex = 0
+			}
 			startIndex, from = splitPath(from)
 		}
-		for i := startIndex; i < endIndex; i++ {
+		for i := startIndex; cmp(i); i = inc(i) {
```

Comment:
What about storing a step instead of introducing a function? `i += step`, which is either `1` or `-1`.
I believe, the condition could be rewritten in a function-less way too `startIndex <= i && i < endIndex` but there are some boundary problems currently.